spec.bs

```diff
@@ -101,3 +230,60 @@ dictionary AuctionAdConfig {
   AbortSignal? signal;
 };
 </xmp>
+
+
+# Structures # {#structures}
+
+<h3 dfn-type=dfn>Interest group</h3>
+
+An interest group is a [=struct=] with the following items:
+
+<dl dfn-for="interest group">
+: <dfn>expiry</dfn>
+:: A point in time at which the browser will forget about this interest group.
+: <dfn>owner</dfn>
+:: An [=url/origin=].
+: <dfn>name</dfn>
+:: A [=string=].
+: <dfn>priority</dfn>
+:: A double.
+: <dfn>enable bidding signals prioritization</dfn>
+:: A [=boolean=].
+: <dfn>priority vector</dfn>
+:: Null or an [=ordered map=] whose [=map/key|keys=] are [=strings=] and whose [=map/value|values=]
+  are doubles.
+: <dfn>priority signals overrides</dfn>
+:: Null or an [=ordered map=] whose [=map/key|keys=] are [=strings=] and whose [=map/value|values=]
+  are doubles.
+: <dfn>execution mode</dfn>
+:: An {{WorkletExecutionMode}}, defaulting to "compatibility".
+: <dfn>bidding url</dfn>
+:: Null or a [=URL=].
+: <dfn>bidding wasm helper url</dfn>
+:: Null or a [=URL=].
+: <dfn>daily update url</dfn>
+:: Null or a [=URL=].
+: <dfn>trusted bidding signals url</dfn>
+:: Null or a [=URL=].
+: <dfn>trusted bidding signals keys</dfn>
+:: Null or a [=list=].
+: <dfn>user bidding signals</dfn>
+:: Null or a [=string=].
+: <dfn>ads</dfn>
+:: Null or a [=list=] of [=interest group ad=].
+: <dfn>ad components</dfn>
+:: Null or a [=list=] of [=interest group ad=].

```

Comment:
Add in more short descriptions of what each field means.

Reply:
Here is what we had previously, when we started with the infra specs.   Not saying this is correct now, however:

```
An {{AuctionAd}} contains the following:

* A <dfn for=AuctionAd dfn-type="dict-member">renderUrl</dfn>, a [=URL=].
* An optional <dfn for=AuctionAd dfn-type="dict-member">metadata</dfn> which is text containing opaque JSON encoded data.


An {{AuctionAdInterestGroup}} is a [=dictionary=] with the following:

* An <dfn export for="AuctionAdInterestGroup">expiration</dfn>, a point in time.

* An <dfn export for="AuctionAdInterestGroup">owner</dfn>, an [=origin=].

* A <dfn export for="AuctionAdInterestGroup">name</dfn>, a [=string=].

* A <dfn export for="AuctionAdInterestGroup">priority</dfn>, a {{double}}, initially 0.

* An <dfn export for="AuctionAdInterestGroup">enableBiddingSignalsPrioritization</dfn>, a [=boolean=], initially false.

* An optional <dfn export for="AuctionAdInterestGroup">priorityVector</dfn>, a [=map=] from [=strings=] to {{double}}.

* An optional <dfn export for="AuctionAdInterestGroup">prioritySignalsOverrides</dfn>, a [=map=] from [=strings=] to {{double}}.

* An <dfn export for="AuctionAdInterestGroup">execution_mode</dfn>, a {{WorkletExecutionMode}}.

* An optional <dfn export for="AuctionAdInterestGroup">biddingLogicURL</dfn>, a [=URL=].

* An optional <dfn export for="AuctionAdInterestGroup">biddingWasmHelperUrl</dfn>, a [=URL=].

* An optional <dfn export for="AuctionAdInterestGroup">dailyUpdateUrl</dfn>, a [=URL=].

* An optional <dfn export for="AuctionAdInterestGroup">trustedBiddingSignalsUrl</dfn>, a [=URL=].

* A <dfn export for="AuctionAdInterestGroup">trustedBiddingSignalsKeys</dfn>, a [=list=] of [=strings=].

* An optional <dfn export for="AuctionAdInterestGroup">userBiddingSignals</dfn>, a [=string=].

* An <dfn export for="AuctionAdInterestGroup">ads</dfn>, a [=list=] of {{AuctionAd}}.

* An <dfn export for="AuctionAdInterestGroup">adComponents</dfn>, a [=list=] of {{AuctionAd}}.
* ```

Reply:
Added some short description for each feild. Let me know if any of those is not clear or needs changes.